source/Interfaces/NetWorth/client/ClientInterface.hpp

```diff
@@ -0,0 +1,161 @@
+             * @param port Destination port number
+             */
+            ClientInterface(const std::string & ip_string,
+                            unsigned short port) {
+                mIp = sf::IpAddress(ip_string);
+                mPort = port;
+            }
+
+            /**
+             * Default destructor
+             */
+            ~ClientInterface() = default;
+
+            /**
+             * Establish connection with server
+             * @return True if successful, false if error
+             */
+            bool EstablishConnection() {
+                // send ping to server
+                sf::Packet send_pkt;
+                sf::Packet recv_pkt;
+                std::string str = "Ping!";
+                send_pkt << str;
+
+                if (mSocket.send(send_pkt, mIp, mPort) != Socket::Status::Done) {
+                    std::cout << "Could not connect to " << mIp << " at port " << mPort << std::endl;
+                    return false;
+                }
+
+                // receive pong from server
+                if (mSocket.receive(recv_pkt, mIp, mPort) != sf::Socket::Done)
+                {
+                    std::cout << "Failure to receive" << std::endl;
+                    return false;
+                }
+                recv_pkt >> str;
+                std::cout << str << std::endl;
+
+                return true;
+            }
+
+            /**
+             * Game loop
+             */
+            void Loop() {
+                char input;
+                bool valid_input = false;
+                bool wait_for_input = true;
+
+                sf::Packet recv_pkt;
+                std::string recv_str;
+                sf::Packet send_pkt;
+
+                recv_pkt >> recv_str;
+                std::cout << recv_str;
+
+                WorldGrid grid;
+                type_options_t type_options;
+                item_set_t item_set;
+                agent_set_t agent_set;
+                std::string action;
+
+                std::string send_str = "Game started";
+                send_pkt << send_str;
+
+                // ask for map
+                if (mSocket.send(send_pkt, mIp, mPort) != Socket::Status::Done) {
+                    std::cout << "Could not connect to " << mIp << " at port " << mPort << std::endl;
+                    return;
+                }
+
+                while (action != "quit")
+                {
+                    // receive initial map
+                    if (mSocket.receive(recv_pkt, mIp, mPort) != sf::Socket::Done)
+                    {
+                        std::cout << "Failure to receive" << std::endl;
+                        return;
+                    }
+                    recv_pkt >> recv_str;
+                    std::cout << std::endl << recv_str;
+
+                    valid_input = false;
+                    while (!valid_input) {
+                        do {
+                            std::cin >> input;
```

Comment:
I don't think you need a loop around this `std::cin`; while we certainly want to move to a real-time interface, `cin` should wait for user input before continuing.